schema_salad/codegen.py

```diff
@@ -89,7 +89,16 @@ def codegen(lang,             # type: str
 
                 cg.declare_field(fieldpred, tl, f.get("doc"), optional)
 
-            cg.end_class(rec["name"])
+            field_names = []
+            for f in rec.get("fields", []):
+                jld = f.get("jsonldPredicate")
+                name = f["name"]
+                if isinstance(jld, dict):
+                    if "_id" in jld and jld["_id"][0] != "@":
+                        name = jld["_id"]
```

Comment:
Why is this here?  I don' think this is correct.  The `_id` field here is used for mapping to RDF, but is not the field name that appears in the CWL document, that is just `f["name"]`.